### PR TITLE
Remove execution_date and logical_date from arguments in api_connexion

### PR DESCRIPTION
**Pull request by sunank200**
This PR removes the `execution_date` and `logical_date` arguments from `api_connexion` that are used to retrieve DAG runs, aligning with the broader changes introduced in Airflow 2.2 and preparing for Airflow 3.0. The functions now use `run_id` as the sole identifier for DAG runs, simplifying the process and eliminating deprecated behaviour.

### **Motivation**:
In Airflow, `execution_date` has historically been used to distinguish different DAG run instances. However, the introduction of `run_id` and the DAG run concept in Airflow 2.2 shifts away from using `execution_date` as an identifier. Continuing to rely on `execution_date` introduces limitations, such as the inability to handle multiple DAG runs at the same logical time, especially in cases like `TriggerDagRunOperator` when dynamic runs are generated.

By removing `execution_date` in favor of `run_id`, this PR eliminates these limitations. This also removes the unique constraint on `execution_date` at the database level, paving the way for a cleaner and more flexible scheduling system in Airflow 3.0.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Are we going to remove it first and then move the API ? I was under the impression we wanted to make it new "while" moving it to the new fast_api framework? But maybe there is a good reason to do it first here?

**Comment by Lee-W** [comment]
> Are we going to remove it first and then move the API ? I was under the impression we wanted to make it new "while" moving it to the new fast_api framework? But maybe there is a good reason to do it first here?

As long as there's no conflict, I think it's ok to do it first 🤔 so that whoever's working on moving the API

File: airflow/models/dagrun.py [hunk 1]
```diff
@@ -285,7 +285,7 @@ def stats_tags(self) -> dict[str, str]:
         return prune_dict({"dag_id": self.dag_id, "run_type": self.run_type})
 
     @property
-    def logical_date(self) -> datetime:
+    def logical_date(self):
         return self.execution_date
```
Comment (Lee-W) [inline airflow/models/dagrun.py:289]: I think this is going to be removed in another PR?

File: airflow/api_connexion/schemas/dag_run_schema.py [hunk 2]
```diff
@@ -109,9 +101,9 @@ def autogenerate(self, data, **kwargs):
 
     @post_dump
     def autofill(self, data, **kwargs):
-        """Populate execution_date from logical_date for compatibility."""
+        """Populate logical_date from logical_date for compatibility."""
         ret_data = {}
-        data["execution_date"] = data["logical_date"]
+        data["logical_date"] = data["logical_date"]
```
Comment (Lee-W) [inline airflow/api_connexion/schemas/dag_run_schema.py:106]:
```suggestion
```

I think we probably don't need this line?

File: airflow/api_connexion/schemas/task_instance_schema.py [hunk 3]
```diff
@@ -212,8 +212,8 @@ class SetTaskInstanceStateFormSchema(Schema):
     @validates_schema
     def validate_form(self, data, **kwargs):
         """Validate set task instance state form."""
-        if not exactly_one(data.get("execution_date"), data.get("dag_run_id")):
-            raise ValidationError("Exactly one of execution_date or dag_run_id must be provided")
+        if not exactly_one(data.get("logical_date"), data.get("dag_run_id")):
```
Comment (Lee-W) [inline airflow/api_connexion/schemas/task_instance_schema.py:215]: Is this still needed after the removal in airflow/api_connexion/openapi/v1.yaml

File: airflow/api_connexion/endpoints/dag_run_endpoint.py [hunk 4]
```diff
@@ -173,12 +173,12 @@ def _fetch_dag_runs(
         query = query.where(DagRun.updated_at <= updated_at_lte)
 
     total_entries = get_query_count(query, session=session)
-    to_replace = {"dag_run_id": "run_id", "execution_date": "logical_date"}
+    to_replace = {"dag_run_id": "run_id", "logical_date": "logical_date"}
```
Comment (Lee-W) [inline airflow/api_connexion/endpoints/dag_run_endpoint.py:176]: Do we need to replace it?

**Comment by Lee-W** [comment]
For these pull requests, we will probably need the `legacy ui` and `legacy api` labels to run the CI. Please let me know if you need help adding the labels to the following pull requests. 🙂

**Comment by sunank200** [comment]
This can be closed rename of `execution_date` is done as part of `logical_date` here: [43902](https://github.com/apache/airflow/pull/43902)

Removal of execution_date is being done in [42404](https://github.com/apache/airflow/pull/42404)
